### PR TITLE
add architecture diagram

### PR DESCRIPTION
**Pull request by stevemar**
I created a draft architecture diagram, please verify for correctness, our graphics team will clean it up. I also snuck in one or two style fixes.

### CODE REVIEW

**Comment by creitz** [comment]
Looks good, except we will need to include AlphaVantage (for stock prices)

File: README.md [hunk 1]
```diff
@@ -2,6 +2,15 @@
 
 # Watson Stock Advisor
 
+![architecture](docs/architecture.png)
+
+## Flow
+1. The user interacts with the backend server via the app UI. Here they are able to add and remove stocks they are interested in.
+2. User input is processed and routed to the backend server, which in turn uses Watson Discovery and a Cloudant NoSQL database. 
+3. The backend server stores stock information in a Cloudant NoSQL database for easy retrieval.
+4. The backend server uses the Watson Discovery  to find information about a specific company.
```
Comment (creitz) [inline README.md]: extra space between 'Discovery' and 'to'

**Review by creitz — APPROVED** [review]

**Comment by stevemar** [comment]
@creitz good call! i'll add something

File: README.md (hunk 1 above)
Comment (stevemar) [reply README.md]: fixed

**Comment by stevemar** [comment]
@creitz see latest revision -- please ignore that the icon is a bit small, the graphics designers will clean it up.

**Pull request merged**